### PR TITLE
fix: doc review.

### PR DESCRIPTION
**Pull request by billy-the-fish**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,27 +41,37 @@ To get the big picture, read [PostgreSQL Hybrid Search Using pgvector](https://w
...
+   psql -d "postgres://<username>:<password>@<host>:<port>/<database-name>"
+   ```
+   
+1. Create the pgai extension:
 
     ```sql
     CREATE EXTENSION IF NOT EXISTS ai CASCADE;
     ```
 
    The `CASCADE` automatically installs the plpython3u and pgvector dependencies.
```
Comment (cevian) [inline README.md:70]: These steps are common to all installation methods, so I would move that down to the common usage section below.

File: README.md [hunk 2]
```diff
@@ -84,63 +91,70 @@ To enable pgai:
...
+* [Run AI queries securely by passing your API key implicitly as a session parameter](#run-ai-queries-securely-by-passing-your-api-key-implicitly-as-a-session-parameter)
+* [Run AI queries securely by passing your API key explicitly as a function argument](#run-ai-queries-securely-by-passing-your-api-key-explicitly-as-a-function-argument)
 
-```bash
-PGOPTIONS="-c ai.openai_api_key=$OPENAI_API_KEY" psql -d "postgres://<username>:<password>@<host>:<port>/<database-name>"
-```
-The `ai.openai_api_key` parameter will be set for the duration of your psql 
-session, and you therefore do not need to specify the `_api_key` parameter to
-pgai functions.
+### Run AI queries securely by passing your API key implicitly as a session parameter
```
Comment (cevian) [inline README.md:113]:
```suggestion
#### Run AI queries securely by passing your API key implicitly as a session parameter
```

File: README.md [hunk 3]
```diff
@@ -84,63 +91,70 @@ To enable pgai:
...
+    `ai.openai_api_key` is set for the duration of your psql session, you do not need to specify it for pgai functions.
+
+    ```sql
+    SELECT * 
+    FROM openai_list_models()
+    ORDER BY created DESC
+    ;
+    ```
+
+### Run AI queries securely by passing your API key explicitly as a function argument
```
Comment (cevian) [inline README.md:139]:
```suggestion
#### Run AI queries securely by passing your API key explicitly as a function argument
```

File: docs/advanced.md [hunk 4]
```diff
@@ -10,20 +10,19 @@ will use pgai to embed, moderate, and summarize git commit history.
 
 ## Load the sample data
 
-To add the advanced examples to your developer environment:
+To add the advanced examples to your developer environment, in the `<pgai-repo>/docs` folder:
```
Comment (cevian) [inline docs/advanced.md:13]: I don't get this change

**Review by cevian — COMMENTED** [review]
Some suggestions

File: README.md (hunk 1 above)
Comment (billy-the-fish) [reply README.md:70]: To create the pgai extension you have to login to the database before you run CREATE command. If I leave out the connection string, our beloved user will have a "HUH?" moment. I'll leave as is.

File: docs/advanced.md (hunk 4 above)
Comment (billy-the-fish) [reply docs/advanced.md:13]: So the user knows they have to be in the basedir of this repo cloned on their local machine.

**Review by jgpruitt — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -84,63 +91,70 @@ To enable pgai:
...
     ```
 
-   The `CASCADE` automatically installs the plpython3u and pgvector dependencies.
+   `CASCADE` automatically installs the plpython3u and pgvector dependencies.
 
-You now [Use pgai with your API keys](#use-pgai-with-your-api-keys) and [Try out the AI models](#usage).
+You now [Connect to your AI provider through pgai](#connect-to-your-ai-provider-through-pgai) and [Try out the AI models](#usage).
 
-## Use pgai with your API keys
+## Connect to your AI provider through pgai
```
[the remark this answers is not in the document]
Comment (billy-the-fish) [reply README.md]: Changed to Use pgai.

File: README.md [hunk 6]
```diff
@@ -84,63 +91,70 @@ To enable pgai:
...
+* [Run AI queries securely by passing your API key implicitly as a session parameter](#run-ai-queries-securely-by-passing-your-api-key-implicitly-as-a-session-parameter)
+* [Run AI queries securely by passing your API key explicitly as a function argument](#run-ai-queries-securely-by-passing-your-api-key-explicitly-as-a-function-argument)
 
-```bash
-PGOPTIONS="-c ai.openai_api_key=$OPENAI_API_KEY" psql -d "postgres://<username>:<password>@<host>:<port>/<database-name>"
-```
-The `ai.openai_api_key` parameter will be set for the duration of your psql 
-session, and you therefore do not need to specify the `_api_key` parameter to
-pgai functions.
+#### Run AI queries securely by passing your API key implicitly as a session parameter
```
[the remark this answers is not in the document]
Comment (billy-the-fish) [reply README.md]: Removed

File: README.md [hunk 7]
```diff
@@ -107,10 +108,10 @@ Most pgai functions require an [OpenAI API key](https://platform.openai.com/docs
...
-* [Run AI queries securely by passing your API key implicitly as a session parameter](#run-ai-queries-securely-by-passing-your-api-key-implicitly-as-a-session-parameter)
-* [Run AI queries securely by passing your API key explicitly as a function argument](#run-ai-queries-securely-by-passing-your-api-key-explicitly-as-a-function-argument)
+* [Run AI queries by passing your API key implicitly as a session parameter](#run-ai-queries-by-passing-your-api-key-implicitly-as-a-session-parameter)
+* [Run AI queries by passing your API key explicitly as a function argument](#run-ai-queries-by-passing-your-api-key-explicitly-as-a-function-argument)
 
-#### Run AI queries securely by passing your API key implicitly as a session parameter
+#### Run AI queries by passing your API key implicitly as a session parameter
 
 To use a [session level parameter when connecting to your database with psql](https://www.postgresql.org/docs/current/config-setting.html#CONFIG-SETTING-SHELL)
 to securely connect to OpenAI through pgai:
```
Comment (jgpruitt) [inline README.md:117]:
```suggestion
to run your AI queries:
```

File: README.md [hunk 8]
```diff
@@ -43,6 +43,7 @@ Before you start working with pgai, you need:
 
 * An [OpenAI API Key](https://platform.openai.com/api-keys) - everyone
 * A postgres client like [psql v16](https://docs.timescale.com/use-timescale/latest/integrations/query-admin/psql/) or [PopSQL](https://docs.timescale.com/use-timescale/latest/popsql/)
+* [Docker](https://docs.docker.com/get-docker/) - if you prefer to use pgai locally
 * [Python3](https://www.python.org/downloads/) - if you prefer code to pure sql 
 
 ## Create a pgai environment
```
Comment (jgpruitt) [inline README.md:49]:
```suggestion
## Installation
```

File: README.md [hunk 9]
```diff
@@ -29,7 +29,7 @@ Timescale offers the following AI journeys:
 
 * **Everyone**: use AI directly from SQL on your data.
   * [Create a pgai environment](#create-a-pgai-environment)
```
Comment (jgpruitt) [inline README.md:31]:
```suggestion
  * [Installation](#installation)
```

File: README.md [hunk 10]
```diff
@@ -95,7 +96,7 @@ To enable pgai:
 
 You now [Connect to your AI provider through pgai](#connect-to-your-ai-provider-through-pgai) and [Try out the AI models](#usage).
 
-## Connect to your AI provider through pgai
+## Use pgai
```
Comment (jgpruitt) [inline README.md:99]:
```suggestion
## Provide your API key to pgai
```

File: README.md [hunk 11]
```diff
@@ -29,7 +29,7 @@ Timescale offers the following AI journeys:
 
 * **Everyone**: use AI directly from SQL on your data.
   * [Create a pgai environment](#create-a-pgai-environment)
-  * [Connect to your AI provider through pgai](#connect-to-your-ai-provider-through-pgai)
+  * [Use pgai](#use-pgai)
```
Comment (jgpruitt) [inline README.md:32]:
```suggestion
  * [Provide your API key to pgai](#provide-your-api-key-to-pgai)
```

File: README.md (hunk 10 above)
Comment (jgpruitt) [reply README.md:99]: Trying to differentiate between this section and "Usage" section.

File: README.md (hunk 8 above)
Comment (jgpruitt) [reply README.md:49]: First thing I look for in a new-to-me repo as a developer/user is an "Installation" section.

File: README.md [hunk 12]
```diff
@@ -10,29 +10,26 @@
...
-* Create OpenAI [embeddings](#openai_embed). 
-* Retrieve OpenAI [chat completions](#openai_chat_complete) from 
+* Create OpenAI [embeddings](#embed). 
+* Retrieve OpenAI [chat completions](#chat_complete) from 
   models such as GPT4o.
-* Facilitate use cases such as [classification, summarization, and data enrichment](#openai_moderate) on your existing 
+* Facilitate use cases such as [classification, summarization, and data enrichment](#moderate) on your existing 
   relational data.
 
 Timescale offers the following AI journeys:
```
Comment (jgpruitt) [inline README.md:28]:
```suggestion
The fastest ways to run PostgreSQL with pgai are:
```

**Review by jgpruitt — CHANGES_REQUESTED** [review]

**Review by jgpruitt — APPROVED** [review]

**Pull request merged**